scripts/minikube.sh

```diff
@@ -104,11 +112,25 @@ up)
         install_kubectl
     fi
 
-    enable_psp
+    MINIKUBE_MAJOR=$(minikube_version 1)
+    MINIKUBE_MINOR=$(minikube_version 2)
+    MINIKUBE_PATCH=$(minikube_version 3)
 
     echo "starting minikube with kubeadm bootstrapper"
-    # shellcheck disable=SC2086
-    minikube start --memory="${MEMORY}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --vm-driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}" ${EXTRA_CONFIG}
+    if [[ "${MINIKUBE_MAJOR}" -ge 1 ]] && [[ "${MINIKUBE_MINOR}" -ge 11 ]] && [[ "${MINIKUBE_PATCH}" -ge 1 ]] || [[ "${MINIKUBE_MAJOR}" -ge 1 ]] && [[ "${MINIKUBE_MINOR}" -ge 12 ]]; then
```

Comment:
Please handle this in the `enable_psp` function, that is much cleaner.

Reply:
If an additional check in enable_psp again I need to check here to handle minikube start, as it will be different for v16.2 to v1.11.0 and >v1.11.1, Handling it in enable_psp might not make sense, this is mainly related to starting a minikube

Reply:
I do not like the current situation. This huge if-statement is very hard to read. Can you at least place that in a function, like
```
# minikube v16.2 to v1.11.0 needs to be restarted to enable PSP
function enabling_psp_needs_restart() {
    local MINIKUBE_MAJOR=...
    ...
    if ...; then
        return 1
    fi
    return 0
}
```

Reply:
With the suggested approach still I need to do the same if check,  not only restart we need to skip passing the psp flag to the minikube start, if you look at the minikube start it's different in both if and else case.

Reply:
Let me know if you prefer to move same check to a new function.

Reply:
Yes, please move it to a function to make it more maintainable.

Feel free to name the function to something more suitable, maybe `need_psp_workaround` or `minikube_supports_psp`?

Reply:
Moved it to a new function